test/unit/mem/buf/src/BufTest.cpp

```diff
@@ -18,6 +18,91 @@
 #include <numeric>
 #include <type_traits>
 
+template<typename TAcc>
+static constexpr auto isAsyncBufferSupported() -> bool
+{
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevCudaRt>)
+    {
+#    if BOOST_LANG_CUDA < BOOST_VERSION_NUMBER(11, 2, 0)
+        return false;
+#    else
+        if constexpr(alpaka::Dim<TAcc>::value > 1)
+        {
+            return false;
+        }
+#    endif // BOOST_LANG_CUDA < BOOST_VERSION_NUMBER(11, 2, 0)
+    }
+#endif // ALPAKA_ACC_GPU_CUDA_ENABLED
+
+#ifdef ALPAKA_ACC_GPU_HIP_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevHipRt>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_GPU_HIP_ENABLED
+
+#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevOacc>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_ANY_BT_OACC_ENABLED
+
+#ifdef ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevOmp5>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+
+    return true;
+}
+
+template<typename TAcc>
+[[maybe_unused]] static auto printAsyncBufferUnsupported() -> void
+{
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevCudaRt>)
+    {
+#    if BOOST_LANG_CUDA < BOOST_VERSION_NUMBER(11, 2, 0)
+        std::cout << "Support for stream-ordered memory buffers require CUDA 11.2 or higher. "
+                  << "The test will be skipped." << std::endl;
+#    else
+        if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevCudaRt> and alpaka::Dim<TAcc>::value > 1)
```

Comment:
```suggestion
        if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevCudaRt> && alpaka::Dim<TAcc>::value > 1)
```
The alternative operator keywords are only availabel in MSVC when compiling in standard conformance mode (`/permissive-`). Unfortunately, CMake chooses the non-conforming mode by default. Please avoid them for the sake of broader MSVC support.

Reply:
sigh ...